### PR TITLE
Ensure that we update the epoch of images so that we don't redraw them. 

### PR DESCRIPTION
**Pull request by jrmuizel**
This cleans up the rawtests a little, adds the ability to track blob image requests, adds a test to make sure that we don't request unnecessarily and fixes the test.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/2240)

### CODE REVIEW

**Comment by jrmuizel** [comment]
@mstange this fixes a constant redrawing bug that I ran into with the current blob image invalidation code.

File: wrench/src/rawtest.rs [hunk 1]
```diff
@@ -160,35 +162,179 @@ impl<'a> RawtestHarness<'a> {
...
+        );
+        resources.update_image(
+            blob_img2,
+            ImageDescriptor::new(500, 500, ImageFormat::BGRA8, true),
+            ImageData::new_blob_image(blob::serialize_blob(ColorU::new(59, 50, 150, 255))),
+            Some(rect(100, 100, 100, 100)),
+        );
+
+        let mut builder = DisplayListBuilder::new(self.wrench.root_pipeline_id, layout_size);
+        builder.push_image(
```
Comment (kvark) [inline wrench/src/rawtest.rs]: a lot of duplicated code with `push_image`, let's have a helper?

File: wrench/src/rawtest.rs [hunk 2]
```diff
@@ -363,18 +478,7 @@ impl<'a> RawtestHarness<'a> {
...
-                root_background_color,
-                layout_size,
-                builder.finalize(),
-                false,
-                ResourceUpdates::new(),
-            );
-            self.wrench
-                .api
-                .generate_frame(self.wrench.document_id, None);
+            self.submit_dl(&mut Epoch(0), layout_size, builder, None);
```
Comment (kvark) [inline wrench/src/rawtest.rs]: nice savings 👍

**Comment by glennw** [comment]
@kvark @jrmuizel Is this ready to merge now?

**Comment by kvark** [comment]
yes!
@bors-servo r+

**Comment by glennw** [comment]
Needs a (small) rebase.

**Pull request merged**